packages/typespec-ts/test/unit/uriTemplate.spec.ts

```diff
@@ -0,0 +1,31 @@
+import { assert } from "chai";
+import { emitParameterFromTypeSpec } from "../util/emitUtil.js";
+import { assertEqualContent } from "../util/testUtil.js";
+
+describe("Client definition generation", () => {
+  it("should generate method-level parameter", async () => {
+    const clientDef = await emitParameterFromTypeSpec(
+      `
+    @route("template/{+param}")
+    op template(param: string): void;
```

Comment:
can you add tests to test against the generated Routes and generated helpers?

Reply:
updated the Routes generation.